qcodes/dataset/measurements.py

```diff
@@ -257,11 +262,13 @@ def _unpack_partial_result(
         """
         param, values = partial_result
         try:
-            parameter = self._interdeps._id_to_paramspec[str(param)]
+            parameter = self._interdeps._id_to_paramspec[str_or_register_name(param)]
         except KeyError:
-            raise ValueError('Can not add result for parameter '
-                             f'{param}, no such parameter registered '
-                             'with this measurement.')
+            raise ValueError(
+                "Can not add result for parameter "
+                f"{param}, no such parameter registered "
+                "with this measurement."
```

Comment:
perhaps worth adding str_or_register_name(param) to the exception so that in case of this happens users are reminded that the register_name may be part of why this error happened.